[Runtimes][CMake] Find Shims, OnoneSupport and Concurrency in FindSwi…

edymtt opened the pull request:
…ftCore

This way we will match most of the targets that would be exported via `SwiftCoreTargets.cmake`.

Take the chance to amend the path to SwiftCore library for Windows.

Addresses rdar://151700875

compnerd reviewed (commented):
I wonder if we want to try to share most of the search logic across the platforms. The differences are the `PLATFORM_SUBDIR`, which we can compute per platform or use the computation.

edymtt commented:
I attempted to refactor the existing code putting more emphasis on the search and the target in #82035 -- @compnerd, could you have a peek at that and check if that goes in the direction you have in mind? Unfortunately the diff is a bit noisy

edymtt committed:
[Runtimes][FindSwiftCore] Find Shims, OnoneSupport and Concurrency

This way we will match most of the targets that would be exported via
SwiftCoreTargets.cmake.

Take the chance to expand the documentation of the find module.

Addresses rdar://151700875

edymtt commented:
@swift-ci please test

edymtt commented:
@swift-ci please smoke test macOS

edymtt commented:
@swift-ci please test Linux

edymtt commented:
@swift-ci please smoke test macOS

edymtt commented:
@swift-ci please smoke test macOS

edymtt commented:
@swift-ci please test Windows

edymtt commented:
@swift-ci please test macOS

edymtt commented:
@swift-ci please smoke test

edymtt commented:
@swift-ci please smoke test Linux

compnerd reviewed (commented):
Minor space issues.

edymtt commented:
@swift-ci please smoke test

edymtt commented on Runtimes/Supplemental/cmake/modules/FindSwiftCore.cmake:
```diff
@@ -143,22 +227,73 @@ find_library(SwiftCore_LIBRARY
+      ${SwiftShims_INCLUDE_DIR_HINTS})
+  add_library(swiftShims INTERFACE IMPORTED GLOBAL)
+  set_target_properties(swiftShims PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${SwiftShims_INCLUDE_DIR}/shims")
```
Thanks for the catch!

edymtt commented:
@swift-ci please test Windows
